### PR TITLE
Deploy playbook

### PR DESCRIPTION
**Pull request by dincho**
in scope of https://www.pivotaltracker.com/story/show/158550692

The playbook is not used yet, but it's part of the migration. It's basically copy/paste from epoch repo with some file name changes, peer key files with changed ansible vault password and [db path reset](https://github.com/aeternity/infrastructure/compare/158550692_deploy_playbook?expand=1#diff-0a05600d18acbea9b59fd306c7e25899R61)

Deploy example: https://circleci.com/gh/aeternity/infrastructure/1254

### CODE REVIEW

File: ansible/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      chain:
+        persist: true
+        db_path: "./db"
+      keypair:
+        dir: "keys"
+        password: "{{ vault_keys_password|default('secret') }}"
+      logging:
+        level: "{{ log_level|default('error') }}"
+      mining:
+        expected_mine_rate: "{{ expected_mine_rate|default(300000) }}"
```
Comment (lucafavatella) [inline ansible/deploy.yml]: https://github.com/aeternity/epoch/pull/1323/files#diff-2ad1947fe9b869ffd23a6a86872e65e0R68 ?

File: ansible/templates/epoch.yaml [hunk 2]
```diff
@@ -33,6 +33,7 @@ websocket:
 
 mining:
     autostart: true
+    beneficiary: {{ config.mining.beneficiary }}
```
Comment (lucafavatella) [inline ansible/templates/epoch.yaml]: It looks to me this needs setting - see my other comment.

File: Makefile [hunk 3]
```diff
@@ -1,11 +1,12 @@
 .DEFAULT_GOAL := lint
+DEPLOY_DOWNTIME ?= 0
+DEPLOY_DB_VERSION ?= ""
```
Comment (lucafavatella) [inline Makefile]: I would expect no default here. What is usage of default? (Caller not setting DB dir is a bug in the caller for me.)

**Comment by lucafavatella** [comment]
I understand corresponding de-duplication in epoch repo tracked in https://www.pivotaltracker.com/story/show/158550692

**Comment by dincho** [comment]
> I understand corresponding de-duplication in epoch repo tracked in https://www.pivotaltracker.com/story/show/158550692

Yes, it's the "run X playbook ..." tasks

File: ansible/templates/epoch.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+---
```
Comment (lucafavatella) [inline ansible/templates/epoch.yaml]: I see this the same as epoch 4f3b0f7c - ok

File: Makefile (hunk 3 above)
Comment (dincho) [reply Makefile]: Fair point, removed in 839db7b

File: ansible/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+  vars:
+    project_root: "{{ ansible_env.HOME }}/node"
+    genesis_accounts_path: "{{ project_root }}/data/aecore/.genesis/accounts.json"
+    is_remote_package: false
+    public_ip: "{{ ansible_ssh_host|default(ansible_host)|default(inventory_hostname) }}"
+    local_peer_keys_dir: "files/peer_keys/{{ public_ip }}"
+    datadog_api_url: https://api.datadoghq.com/api/v1/downtime?api_key={{ vault_datadog_api_key }}&application_key={{ vault_datadog_app_key }}
+    downtime: "0"
+    db_version: ""
```
Comment (lucafavatella) [inline ansible/deploy.yml]: ?

Comment (dincho) [reply ansible/deploy.yml]: I amended previous commit so 0249283

File: Makefile (hunk 3 above)
Comment (dincho) [reply Makefile]: Amended to 0249283

File: ansible/group_vars/tag_env_integration/genesis_accounts.yml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+genesis_accounts:
```
Comment (lucafavatella) [inline ansible/group_vars/tag_env_integration/genesis_accounts.yml]: Is this file name meaningful? I saw changed from accounts.yml in epoch repo, though it looks like name does not matter so not reference to it needs update.

File: ansible/mnesia_backup.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (lucafavatella) [inline ansible/mnesia_backup.yml]: Clarify whether this part.

File: ansible/group_vars/tag_env_integration/genesis_accounts.yml (hunk 6 above)
Comment (dincho) [reply ansible/group_vars/tag_env_integration/genesis_accounts.yml]: No, the filename does not matter from execution point of view. I renamed it to have more meaningful name in this repo context/scope.

**Comment by dincho** [comment]
> Just checking - tag_seed_uat inventory group not present?

I didn't migrated it as it's not used by automation, however it should be covered in https://www.pivotaltracker.com/story/show/158872869

File: ansible/mnesia_backup.yml (hunk 7 above)
Comment (dincho) [reply ansible/mnesia_backup.yml]: The file was added by accident. Now removed.

**Pull request merged**